### ISSUE TITLE
need a writeSync function

### ISSUE THREAD

**Issue by li986390329** [sleeyax/asarmor#9]
Question: 
I used the asarmor module in the afterPack process, but I found that 7za stopped processing during the generation of the zip package, so I started looking for the cause of the problem, and finally I found that using the write function would generate a .tmp file, and the write method had not finished when 7za packed the file, causing the 7za compression to report an error:
```
WARNING: No such file or directory
    xxxxx.app/Contents/Resources/app.asar.tmp
```

I took a look at the source code and added a promise to the write method and the problem was solved, but it would be better to provide a method that can be called synchronously, thank you very much.

code like this :
```    
Asarmor.prototype.writeSync = function (output) {
        return new Promise((resolve, reject) => {
            // Convert header back to string
            var headerPickle = pickle.createEmpty();
            headerPickle.writeString(JSON.stringify(this.archive.header));
            // Read new header size
            var headerBuffer = headerPickle.toBuffer();
            var sizePickle = pickle.createEmpty();
            sizePickle.writeUInt32(headerBuffer.length);
            var sizeBuffer = sizePickle.toBuffer();
            // Write everything to output file :D
            var tmp = output + '.tmp'; // create temp file bcs we can't read & write the same file at the same time
            var writeStream = fs_1.default.createWriteStream(tmp, { flags: 'w' });
            writeStream.write(sizeBuffer);
            writeStream.write(headerBuffer);
            // write unmodified contents
            var fd = fs_1.default.openSync(this.filePath, 'r');
            var originalHeaderSize = this.readHeaderSize(fd);
            fs_1.default.closeSync(fd);
            var readStream = fs_1.default.createReadStream(this.filePath, { start: this.headerSizeOffset + originalHeaderSize });
            readStream.pipe(writeStream);
            readStream.on('close', function () { return readStream.unpipe(); });
            writeStream.on('close', () => {
                fs_1.default.renameSync(tmp, output);
                console.log("Asarmor sync write end");
                return resolve();
            });
        })
    };
```

**Comment by sleeyax** [comment]
It needs to RW asynchronously because it has to be able to read very large files, see #4 

However, I didn't take into account additional after-processing on the resulting asar. So it should resolve to a promise indeed, like in your example.

Do you mind creating a PR?

**Comment by li986390329** [comment]
I need some time, and I'll create a  PR when I'm done.

**Comment by klren0312** [comment]
I also need this function to make incremental update
but I think it can modify the original function (write) directly, https://github.com/sleeyax/asarmor/pull/10

**Comment by sleeyax** [comment]
Closed by previously mentioned PR & released in [v1.8.0](https://www.npmjs.com/package/asarmor/v/1.8.0).
